Sources/TSCBasic/TemporaryFile.swift

```diff
@@ -213,14 +237,14 @@ extension MakeDirectoryError: CustomNSError {
 ///            variables TMPDIR, TEMP and TMP will be checked for a value (in that order). If none of the env
 ///            variables are set, dir will be set to `/tmp/`.
 ///     - prefix: The prefix to the temporary file name.
-///     - removeTreeOnDeinit: If enabled try to delete the whole directory tree otherwise remove only if its empty.
 ///     - body: A closure to execute that receives the absolute path of the directory as an argument.
-///             If `body` has a return value, that value is also used as the
-///             return value for the `withTemporaryDirectory` function.
+///           If `body` has a return value, that value is also used as the
+///           return value for the `withTemporaryDirectory` function.
+///           The cleanup block should be called when the temporary directory is no longer needed.
 ///
 /// - Throws: MakeDirectoryError and rethrows all errors from `body`.
 public func withTemporaryDirectory<Result>(
-    dir: AbsolutePath? = nil, prefix: String = "TemporaryDirectory", removeTreeOnDeinit: Bool = false , _ body: (AbsolutePath) throws -> Result
+    dir: AbsolutePath? = nil, prefix: String = "TemporaryDirectory" , _ body: (AbsolutePath, @escaping (AbsolutePath) -> Void) throws -> Result
```

Comment:
can these changes move to a separate PR? or are they related to the async process?

Reply:
They are needed upstream in SPM.

Reply:
I am a bit iffy about the `@escaping (AbsolutePath) -> Void` cleanup argument passed into the closure. feel like the boolean cleanup is good enough for the public API and this should an internal implementation details?

Reply:
My rationale for this is to avoid code duplication.

Reply:
@pcbeard right. not a big deal, mostly wondering if the variant with the cleanup closure needs to be public, or if the boolean variant is good enough as the public api. i think SwiftPM can just use the boolean variant?